# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/ipc/ParentProcessDocumentChannel.h
# Commit: 28bca5c5b8b4
# Full Hash: 28bca5c5b8b4d5a4f8e6ecaf621738ad955c3121
# Author: Jean-Yves Avenard <jyavenard@mozilla.com>
# Date: 2020-06-09 15:16:49
# Regressor Bug: 1641737
# File Overlap Count: 2
# Description:
#   Bug 1641737 - P2. Use MozPromise with DocumentLoadListener and remove cycle. r=mattwoodrow.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D78179
# ==============================================================================

diff -r 0296d0f6c1d1 -r 28bca5c5b8b4 netwerk/ipc/ParentProcessDocumentChannel.h
--- a/netwerk/ipc/ParentProcessDocumentChannel.h	Tue Jun 09 07:23:29 2020 +0000
+++ b/netwerk/ipc/ParentProcessDocumentChannel.h	Tue Jun 09 10:11:11 2020 +0000
@@ -8,7 +8,6 @@
 #define mozilla_net_ParentProcessDocumentChannel_h
 
 #include "ProtocolUtils.h"
-#include "mozilla/net/ADocumentChannelBridge.h"
 #include "mozilla/net/DocumentChannel.h"
 #include "mozilla/net/DocumentLoadListener.h"
 #include "nsIObserver.h"
@@ -18,8 +17,7 @@
 
 class ParentProcessDocumentChannel : public DocumentChannel,
                                      public nsIAsyncVerifyRedirectCallback,
-                                     public nsIObserver,
-                                     public ADocumentChannelBridge {
+                                     public nsIObserver {
  public:
   ParentProcessDocumentChannel(nsDocShellLoadState* aLoadState,
                                class LoadInfo* aLoadInfo,
@@ -37,23 +35,13 @@
   RedirectToRealChannel(
       nsTArray<ipc::Endpoint<extensions::PStreamFilterParent>>&&
           aStreamFilterEndpoints,
-      uint32_t aRedirectFlags, uint32_t aLoadFlags) override;
-
-  void DisconnectChildListeners(nsresult aStatus,
-                                nsresult aLoadGroupStatus) override {
-    DocumentChannel::DisconnectChildListeners(aStatus, aLoadGroupStatus);
-    RemoveObserver();
-    mDocumentLoadListener = nullptr;
-  }
-  void Delete() override {}
+      uint32_t aRedirectFlags, uint32_t aLoadFlags);
   void DeleteIPDL() override {
     mPromise.ResolveIfExists(NS_BINDING_ABORTED, __func__);
   }
-  base::ProcessId OtherPid() const override { return 0; }
 
  private:
   virtual ~ParentProcessDocumentChannel();
-  void DisconnectDocumentLoadListener();
   void RemoveObserver();
 
   RefPtr<DocumentLoadListener> mDocumentLoadListener;
@@ -61,6 +49,7 @@
       mStreamFilterEndpoints;
   MozPromiseHolder<PDocumentChannelParent::RedirectToRealChannelPromise>
       mPromise;
+  MozPromiseRequestHolder<DocumentLoadListener::OpenPromise> mPromiseRequest;
   bool mRequestObserversCalled = false;
 };
 